modules/core/src/test/java/org/apache/ignite/testframework/GridTestUtils.java

```diff
@@ -1798,6 +1766,35 @@ public static boolean waitForCondition(GridAbsPredicate cond, long timeout) thro
         return false;
     }
 
+    /**
+     * Waits until given condition is met or timeout. Exception free check. Polling in busy wait loop is used.
+     *
+     * @param cond condition to met.
+     * @param timeout max time to wait in milliseconds.
+     * @return {@code true} if condition was met, {@code false} otherwise or in case of timeout/exception.
+     */
+    public static boolean waitUntil(Supplier<Boolean> cond, long timeout) {
+        assert timeout > 0;
+        long curTime = U.currentTimeMillis();
+        long endTime = curTime + timeout;
+
+        try {
+            while (curTime < endTime) {
+                if (cond.get()) {
+                    return true;
+                }
+                U.sleep(DFLT_BUSYWAIT_SLEEP_INTERVAL);
+
+                curTime = U.currentTimeMillis();
+            }
+        }
+        catch (Exception ex) {
+            return false;
+        }
+
+        return false;
```

Comment:
This can lead to false-negative result as condition wasn't check after last sleep.
Should be "return con.get()"